assert/assertions.go

```diff
@@ -165,19 +165,40 @@ func ObjectsAreEqualValues(expected, actual interface{}) bool {
 		return true
 	}
 
-	actualType := reflect.TypeOf(actual)
-	if actualType == nil {
+	expectedValue := reflect.ValueOf(expected)
+	actualValue := reflect.ValueOf(actual)
+	if !expectedValue.IsValid() || !actualValue.IsValid() {
 		return false
 	}
-	expectedValue := reflect.ValueOf(expected)
-	if expectedValue.IsValid() && expectedValue.Type().ConvertibleTo(actualType) {
+
+	expectedType := expectedValue.Type()
+	actualType := actualValue.Type()
+	if expectedType.ConvertibleTo(actualType) {
+		if isNumericType(expectedType) && isNumericType(actualType) {
+			// If both values are numeric, there are chances of false positives due
+			// to overflow or underflow. So, we need to make sure to always convert
+			// the smaller type to a larger type before comparing.
+			if expectedType.Size() >= actualType.Size() {
+				return actualValue.Convert(expectedType).Interface() == expected
+			}
+
+			return expectedValue.Convert(actualType).Interface() == actual
+		}
+
 		// Attempt comparison after type conversion
 		return reflect.DeepEqual(expectedValue.Convert(actualType).Interface(), actual)
 	}
 
 	return false
```

Comment:
If it is possible, I would prefer to invert the statements to reduce the indenting?

It makes it easier for me to read (but it doesn't look like it is a practice within the project)

Reply:
Makes sense. will do that and push the changes in a bit.

Reply:
Done